src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource.cs

```diff
@@ -69,7 +70,7 @@ private sealed partial class TagSource : ForegroundThreadAffinitizedObject
             /// Series of tokens used to cancel previous outstanding work when new work comes in. Also used as the lock
             /// to ensure threadsafe writing of _eventWorkQueue.
             /// </summary>
-            private readonly CancellationSeries _cancellationSeries;
+            private readonly ReferenceCountedDisposable<CancellationSeries> _cancellationSeries;
```

Comment:
it very much feels like it takse a bunch of extra tuff just to properly handle cancellation lifetime :(

Reply:
When the object is both disposable and asynchronous, yes.